hack/test_errata.py

```diff
@@ -0,0 +1,598 @@
+#         {
+#             "msg": {
+#                 "product": "RHOSE",
+#                 "to": "SHIPPED_LIVE",
+#             }
+#         }
+#     ]
+#     json_load_mock.return_value = {
+#         "raw_messages": raw_messages,
+#         "pages": 1
+#     }
+    
+#     # Retrive messages from errata.poll without specifying parametres
+#     polled_messages = []
+#     for message in errata.poll():
+#         polled_messages.append(message)
+
+#     # Test if data was polled correctly
+#     expected_messages = [x['msg'] for x in raw_messages]
+#     assert expected_messages == polled_messages
+
+
+@patch("json.load")
+@patch("urllib.request.urlopen")
+def test_poll_number_of_returned_pages_is_zero(urlopen_mock, json_load_mock):
+    urlopen_mock.return_value = MagicMock()
+
+    raw_messages = []
+    json_load_mock.return_value = {
+        "raw_messages": raw_messages,
+        "pages": 0
+    }
+
+    polled_messages = []
+    poll_period = datetime.timedelta(seconds=3600)
+    for message in errata.poll(period=poll_period):
+        polled_messages.append(message)
+
+    # Test if data was polled correctly
+    assert polled_messages == []
+
+
+@patch("json.load")
+@patch("urllib.request.urlopen")
+def test_poll_no_results(urlopen_mock, json_load_mock):
+    urlopen_mock.return_value = MagicMock()
+
+    raw_messages = []
+    json_load_mock.return_value = {
+        "raw_messages": raw_messages,
+        "pages": 1
+    }
+
+    polled_messages = []
+    poll_period = datetime.timedelta(seconds=3600)
+    for message in errata.poll(period=poll_period):
+        polled_messages.append(message)
+    
+    # Test if data was polled correctly
+    assert polled_messages == []
+
+
+@patch("json.load")
+@patch("time.sleep")
+@patch("urllib.request.urlopen")
+def test_poll_unresponsive_url_becomes_responsive(urlopen_mock, sleep_mock, json_load_mock):
+    # First time calling request.urlopen returns Exception, 
+    # second time returns mocked HTTPResponse
+    urlopen_mock.side_effect = [
+    Exception("Unresponsive, request.urlopen has failed"), MagicMock()]
+    raw_messages = [
+        {
+            "msg": {
+                "product": "RHOSE",
+                "to": "SHIPPED_LIVE",
+            }
+        }
+    ]
+    json_load_mock.return_value = {
+        "raw_messages": raw_messages,
+        "pages": 1
+    }
+    
+    # Retrive messages from errata.poll
+    poll_period = datetime.timedelta(seconds=3600)
+    polled_messages = []
+    for message in errata.poll(period=poll_period):
+        polled_messages.append(message)
+
+    # Test if data was polled correctly
+    expected_messages = [x['msg'] for x in raw_messages]
+    assert expected_messages == polled_messages
+
+    # URL wasn't responsive only once, so sleep should be called only once
+    sleep_mock.assert_called_once()
+
+
+@patch("json.load")
+@patch("urllib.request.urlopen")
+def test_poll_multiple_messages(urlopen_mock, json_load_mock):
```

Comment:
Picking a random line so we can mark this thread resolved once it is, but one of the issues has been our regexp not matching the synopses we've seen in the wild.  Working through `git log hack/errata.py`, we want to match:

* OpenShift Container Platform 4.6 GA Images
* Moderate: OpenShift Container Platform 4.5.20 bug fix and golang security update
* Low: OpenShift Container Platform 4.3.40 security and bug fix update

while excluding:

* OpenShift Container Platform 4.5.11 optional CSI driver Operators bug fix update

Probably worth mining more of DataGrepper's history or the cache of a running errata pod for more examples to feed into the synopsis-matching test.